src/passes/transform/isolate_primitives.cpp

```diff
@@ -0,0 +1,84 @@
+#include "coreir/passes/transform/isolate_primitives.h"
+
+std::string CoreIR::Passes::IsolatePrimitives::ID = "isolate_primitives";
+
+// Creates a separate module that isolates all the primitive (coreir/corebit) instances
+
+namespace CoreIR {
+namespace Passes {
+bool IsolatePrimitives::runOnModule(Module* m) {
+  if (!m->hasDef()) {return false;}
+  auto def = m->getDef();
+  auto c = this->getContext();
+
+  //get a map of only primitive instances
+  auto primitiveInstances = filterOver(
+    def->getInstances(),
+    [](auto it) {
+      auto nsname = it.second->getModuleRef()->getNamespace()->getName();
+      return nsname == "coreir" || nsname == "corebit";
+    }
+  );
+
+  //early out
+  if (primitiveInstances.size()==0) {return false;}
+
+  std::vector<std::pair<Wireable*, Wireable*>> internal_connections; //prim <=> prim
+  std::vector<std::pair<Wireable*, Wireable*>> external_connections; //prim <=> other
+  for (auto const& [wA, wB] : def->getConnections()) {
+    //Put into two piles external and internal
+    bool AIsPrimitive = primitiveInstances.count(wA->getSelectPath()[0]);
+    bool BIsPrimitive = primitiveInstances.count(wB->getSelectPath()[0]);
+    if (AIsPrimitive && BIsPrimitive) {internal_connections.push_back(std::make_pair(wA, wB));}
+    else if (AIsPrimitive) {external_connections.push_back(std::make_pair(wA, wB));}
+    else if (BIsPrimitive) {external_connections.push_back(std::make_pair(wB, wA));}
+  }
+
+  //Construct module type interface
```

Comment:
Standalone comments should be `// start of comment` and comments at the end of line should be like `my_code_here();  // start of comment`.

I think running clang-format will fix all of these.